heapster-saw/src/Verifier/SAW/Heapster/HintExtract.hs

```diff
@@ -49,12 +49,14 @@ type ExtractM = Except String
 extractHints ::
   forall ghosts args outs blocks init ret.
   PermEnv ->
-  [L.Module] ->
-  -- ^ The original source modules: used for finding constant values (i.e. spec strings)
-  FunPerm ghosts args outs ret ->
-  -- ^ The FunPerm corresponding to the CFG we are scanning
-  CFG LLVM blocks init ret ->
-  -- ^ The Crucible CFG for which to build the block hint map
+  [L.Module]   -- ^ The original source modules: used for finding constant values (i.e. spec strings)
+  ->
```

Comment:
The convention of putting a function arrow on its own line is rather strange. Why not do this instead?

```suggestion
  [L.Module]   {- ^ The original source modules: used for finding constant values (i.e. spec strings) -} ->
```

And similarly for all other occurrences of this convention in this PR.